ksp_plugin_adapter/localization_extensions.cs

```diff
@@ -28,23 +29,51 @@ internal static class L10N {
     return Localizer.Format("#Principia_GrammaticalForm_Standalone", name);
   }
 
-  public static string NameWithoutArticle(this CelestialBody body) {
-    // This will need to be adjusted when we add support for other languages
-    // with articles.
-    return (Localizer.CurrentLanguage == english_us_ &&
-            body.displayName.StartsWith("The ")) ? body.name : body.displayName;
+  public static string StandaloneName(this CelestialBody celestial) {
+    return Standalone(celestial.NameWithoutArticle());
+  }
+
+  private static bool StartsWithCapitalizedDefiniteArticle(string s) {
+    return (Localizer.CurrentLanguage == english_us_ && s.StartsWith("The ")) ||
+           (Localizer.CurrentLanguage == french_ &&
+            (s.StartsWith("La ") || s.StartsWith("Le ")));
+  }
+
+  private static string LingoonaUnqualified(string s) {
+    return s.Split(new[]{'^'}, 2)[0];
+  }
+
+  private static string LingoonaQualifiers(string s) {
+    if (!s.Contains('^')) {
+      return "";
+    }
+    return s.Split(new[]{'^'}, 2)[1];
+  }
+
+  private static string LingoonaQualify(string s, string qualifiers) {
+    return qualifiers == "" ? s : $"{s}^{qualifiers}";
+  }
+
+  private static string NameWithoutArticle(this CelestialBody body) {
+    return StartsWithCapitalizedDefiniteArticle(body.displayName)
+        ? LingoonaQualify(
+              body.displayName.Split(new[]{' '}, 2)[1],
+              LingoonaQualifiers(body.displayName).Replace("d", ""))
+        : body.displayName;
   }
 
   public static string NameWithArticle(this CelestialBody body) {
-    // TODO(egg): should we apped ^d instead of prepending "the"?  Certainly
-    // we would need to do that for other languages.
-    return (Localizer.CurrentLanguage == english_us_ &&
-            body.displayName.StartsWith("The ")) ? "the " + body.name
-                                                 : body.displayName;
+    return StartsWithCapitalizedDefiniteArticle(body.displayName)
+        ? body.displayName.Contains('^')
+              ? body.displayName.Split(new[]{' '}, 2)[1] + "d"
```

Comment:
Extract the `Split` into a local variable, no point in repeating a complicated expression.